### PR TITLE
Remove npm ci production flag and add note

### PR DESCRIPTION
**Pull request by mbareta**

### CODE REVIEW

File: recipes/docker-image-guide.md [hunk 1]
```diff
@@ -133,7 +135,7 @@ CMD [ "node", "index.js" ]
   By copying source code files after dependency installation, we are only
   re-executing those steps that come after that step, including that step.
 
-10. `RUN npm ci --only=production`
+10. `RUN npm ci`
```
Comment (ikovac) [inline recipes/docker-image-guide.md:138]: Does it make sense to also describe here that NODE_ENV needs to be set to production in order to install only production dependencies?

File: recipes/docker-image-guide.md [hunk 2]
```diff
@@ -56,7 +56,9 @@ WORKDIR /usr/src/app
 ENV NODE_ENV production
 COPY package*.json .
 # https://cheatsheetseries.owasp.org/cheatsheets/NodeJS_Docker_Cheat_Sheet.html#2-install-only-production-dependencies-in-the-nodejs-docker-image
-RUN npm ci --only=production
+# when NODE_ENV is set to production, npm ci automatically omits dev dependencies
+# https://docs.npmjs.com/cli/v10/commands/npm-ci#omit
+RUN npm ci
```
Comment (droguljic) [inline recipes/docker-image-guide.md:61]: Here is my two cents. It is always better to be explicit and state your intentions. Mainly for two reasons, clarity and future-proofing against unintended behavior.

Comment (mbareta) [reply recipes/docker-image-guide.md:61]: So, put `--omit=dev` flag with `npm ci`?

Comment (ikovac) [reply recipes/docker-image-guide.md:61]: @mbareta yes

File: recipes/docker-image-guide.md [hunk 3]
```diff
@@ -187,15 +189,15 @@ ENTRYPOINT ["dumb-init", "--"]
 FROM node:20.9-bookworm@sha256:3c48678afb1ae5ca5931bd154d8c1a92a4783555331b535bbd7e0822f9ca8603 AS build
 WORKDIR /usr/src/app
 COPY package*.json .
-RUN npm ci
+RUN npm ci --omit=dev
```
Comment (ikovac) [inline recipes/docker-image-guide.md:192]: In this step, we may want to keep dev dependencies as this is a build step that probably requires some build tools that are available as dev dependencies.

**Review by ikovac — CHANGES_REQUESTED** [review]

File: recipes/docker-image-guide.md (hunk 3 above)
Comment (mbareta) [reply recipes/docker-image-guide.md:192]: You are correct. Building the frontend should either require dev dependencies to be installed in Docker or move build lib (vite, webpack, ...) to prod dependencies.

Comment (mbareta) [reply recipes/docker-image-guide.md]: But also, we should set NODE_ENV to production in this step, import .env, set secrets etc. What I'm saying is, we should add more use cases and details in the future.

**Review by ikovac — APPROVED** [review]

**Pull request merged**